lib/dry/auto_inject/injector.rb

```diff
@@ -2,22 +2,30 @@
 
 module Dry
   module AutoInject
-    class Injector
+    class Injector < BasicObject
       # @api private
       attr_reader :container
 
       # @api private
       attr_reader :strategy
 
       # @api private
-      def initialize(container, strategy)
+      attr_reader :builder
+
+      # @api private
+      def initialize(container, strategy, builder:)
         @container = container
         @strategy = strategy
+        @builder = builder
       end
 
       def [](*dependency_names)
         strategy.new(container, *dependency_names)
       end
+
+      def method_missing(name, *args, &block)
+        builder.send(name)
```

Comment:
Also should implement `respond_to_missing(method, _include_private = false)` when using `method_missing`